hibernate-core/src/main/java/org/hibernate/boot/model/process/spi/MetadataBuildingProcess.java

```diff
@@ -732,6 +730,42 @@ private static void adaptToPreferredSqlTypeCode(
 		// else warning?
 	}
 
+	private static void adaptToPreferredSqlTypeCodeForUuid(
+			TypeConfiguration typeConfiguration,
+			JdbcTypeRegistry jdbcTypeRegistry,
+			int preferredSqlTypeCodeForUuid) {
+		final JavaTypeRegistry javaTypeRegistry = typeConfiguration.getJavaTypeRegistry();
+		final BasicTypeRegistry basicTypeRegistry = typeConfiguration.getBasicTypeRegistry();
+		final BasicType<?> uuidType = new NamedBasicTypeImpl<>(
+				javaTypeRegistry.getDescriptor( UUID.class ),
+				jdbcTypeRegistry.getDescriptor( preferredSqlTypeCodeForUuid ),
+				"uuid"
+		);
+		basicTypeRegistry.register(
+				uuidType,
+				UUID.class.getSimpleName(),
+				UUID.class.getName()
+		);
+	}
+
+	private static void adaptToPreferredSqlTypeCodeForDuration(
+			TypeConfiguration typeConfiguration,
+			JdbcTypeRegistry jdbcTypeRegistry,
+			int preferredSqlTypeCodeForDuration) {
+		final JavaTypeRegistry javaTypeRegistry = typeConfiguration.getJavaTypeRegistry();
+		final BasicTypeRegistry basicTypeRegistry = typeConfiguration.getBasicTypeRegistry();
+		final BasicType<?> durationType = new NamedBasicTypeImpl<>(
+				javaTypeRegistry.getDescriptor( Duration.class ),
+				jdbcTypeRegistry.getDescriptor( preferredSqlTypeCodeForDuration ),
+				"duration"
+		);
+		basicTypeRegistry.register(
+				durationType,
+				Duration.class.getSimpleName(),
+				Duration.class.getName()
+		);
+	}
+
```

Comment:
I don't understand what the difference between these methods is. Looks like they could be one method parameterized by a `Class`, an `int`, and a `String`. 

And then these two and also `adaptToPreferredSqlTypeCodeForInstant()` would go away.